airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionSettingsTab/components/SchemaUpdateNotifications.tsx

```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import { FormattedMessage } from "react-intl";
+import { useAsyncFn } from "react-use";
+
+import { ControlLabels } from "components";
+import { Card } from "components/ui/Card";
+import { Switch } from "components/ui/Switch";
+
+import { useConnectionEditService } from "hooks/services/ConnectionEdit/ConnectionEditService";
+
+import styles from "./SchemaUpdateNotifications.module.scss";
+
+export const SchemaUpdateNotifications: React.FC = () => {
+  const { connection, updateConnection, connectionUpdating } = useConnectionEditService();
+  const [{ loading }, onChange] = useAsyncFn(async (checked: boolean) => {
```

Comment:
`connectionUpdating` handles this loading bit.

Reply:
<Switch
        checked={connection.notifySchemaChanges}
        onChange={(event) => onChange(event.target.checked)}
        
        disabled={connectionUpdating} - should disable `switch` when the user interacts with `useConnectionEditService` 
        form different elements of UI ()
        
        loading={loading} - user will see a loading indicator to interact with the current toggle
      />
      
We can't use just one `connectionUpdating`. It's shared state from different components from this page (ConnectionSettingsTab)

Behavior with `connectionUpdating` for loading state:
loom.com/share/82276b66eef5496c83a1a52ed853370e
It looks buggy :face_with_head_bandage: